statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -1369,7 +1445,28 @@ def untransform_params(self, constrained):
         """
         return np.array(constrained, ndmin=1)
 
-    def update(self, params, transformed=True, complex_step=False):
+    def handle_params(self, params, transformed=True, includes_fixed=False,
```

Comment:
`includes_fixed` is passed through a bunch of different methods, but this seems like the only place where it is used directly.  I'm wondering if the contextmanager call can set some internal state (possibly related to the self._has_fixed_params already used here) that makes the includes_fixed arg unnecessary.

Reply:
Mechanically, the reason is that these methods may be given either a full parameter vector (including both fixed and free parameters) or a subset containing only the free parameters.

The reason why the implementation supports both cases was to avoid rewriting a bunch of code (especially in e.g. the results object) that expects all of the parameters. I should take another look to see if this is really necessary.